### DISCUSSION TITLE
Hot Ones YouTube TVDB Matching

### DISCUSSION THREAD

**Discussion by baxterblk** [RandomNinjaAtk/arr-scripts#295] [Q&A] [answered]
I'm currently manually adding the YouTube Video URLs to each inidivual episode on TVDB for "Hot Ones".

Just want to ensure that I'm adding the URLs to the correct place for the yt-dlp script to be able to download from.

Furthermore, do we know how long it takes the changes to propogate to be recognized by the script(s)?

![image](https://github.com/user-attachments/assets/24a50634-2cbf-49e4-82b9-4ea46bff9204)

**Comment by baxterblk** [comment]
@RandomNinjaAtk

**Reply by RandomNinjaAtk** [reply]
Looks like it is looking at production code:
https://github.com/RandomNinjaAtk/arr-scripts/blob/f9e9cc6d87f1f2b36949d640508e1569aa254aa4/sonarr/YoutubeSeriesDownloader.service#L94

Example page: https://www.thetvdb.com/series/the-slow-mo-guys/episodes/10255746

If you give me an example link to one of the ones you've done, I can look into any changes that may need to take place.

**Reply by baxterblk** [reply, marked as answer]
Here what I'm seeing... which states it grabs for a URL first **then** it looks for production code.

```
# Fetch the TVDB page for the episode
95: tvdbPageData=$(curl -s "https://thetvdb.com/series/$seriesTvdbTitleSlug/episodes/$episodeId")

# Attempt to extract a YouTube URL directly from the TVDB page
96: downloadUrl=$(echo "$tvdbPageData" | grep -i youtube.com  | grep -i watch | grep -Eo "(http|https)://[a-zA-Z0-9./?=_%:-]*")

# If no YouTube URL is found, check for the YouTube network and use the production code
98: if [ -z $downloadUrl ]; then
99:     network="$(echo "$tvdbPageData" | grep -i "/companies/youtube")"

I've updated ALL episodes on TVDB with the corresponding URL to the main website i.e. YouTube URL.
100:    if [ ! -z "$network" ]; then 
101:        downloadUrl=$(echo "$tvdbPageData" | grep -iws "production code" -A 2 | sed 's/\ //g' | tail -n1)
102:        if [ ! -z $downloadUrl ]; then
103:            downloadUrl="https://www.youtube.com/watch?v=$downloadUrl"
104:        fi
105:    fi
106: fi
```

**Reply by baxterblk** [reply]
https://www.thetvdb.com/series/hot-ones/episodes/10132670

**Reply by RandomNinjaAtk** [reply]
> Here what I'm seeing... which states it grabs for a URL first **then** it looks for production code.
> 
> ```
> # Fetch the TVDB page for the episode
> 95: tvdbPageData=$(curl -s "https://thetvdb.com/series/$seriesTvdbTitleSlug/episodes/$episodeId")
> 
> # Attempt to extract a YouTube URL directly from the TVDB page
> 96: downloadUrl=$(echo "$tvdbPageData" | grep -i youtube.com  | grep -i watch | grep -Eo "(http|https)://[a-zA-Z0-9./?=_%:-]*")
> 
> # If no YouTube URL is found, check for the YouTube network and use the production code
> 98: if [ -z $downloadUrl ]; then
> 99:     network="$(echo "$tvdbPageData" | grep -i "/companies/youtube")"
> 
> I've updated ALL episodes on TVDB with the corresponding URL to the main website i.e. YouTube URL.
> 100:    if [ ! -z "$network" ]; then 
> 101:        downloadUrl=$(echo "$tvdbPageData" | grep -iws "production code" -A 2 | sed 's/\ //g' | tail -n1)
> 102:        if [ ! -z $downloadUrl ]; then
> 103:            downloadUrl="https://www.youtube.com/watch?v=$downloadUrl"
> 104:        fi
> 105:    fi
> 106: fi
> ```

Yes, you are right.

**Reply by baxterblk** [reply]
Understood. So I'm just waiting for propogation then it seems.

**Reply by RandomNinjaAtk** [reply]
> Understood. So I'm just waiting for propogation then it seems.

Yeah, I think that is the issue. I modified a couple for some testing, and when I curl the pages, I don't see the info... yet visiting the site, I see the info... Odd... probably due to caching or something else related to that....

**Comment by RandomNinjaAtk** [comment]
@baxterblk Looks like what your doing is working because I'm getting more episodes... Keep up the good work! Thanks!

Real heroes are the ones that take the time to populate/update the databases!

**Reply by baxterblk** [reply]
That's good news!! 🙂

**Reply by baxterblk** [reply]
@RandomNinjaAtk 

I've also added all the links for Specials as well.

I'll need to ensure S01-S03 are up-to-date with correct URLs. I think I started adding at Season 03. 

Tons of work. About 12 hours in total.
